src/Sentry/Internal/Extensions/JsonExtensions.cs

```diff
@@ -551,15 +551,18 @@ private static JsonSerializerContext GetSerializerContext(Type type, bool preser
 
     private static byte[] InternalSerializeToUtf8Bytes(object value)
     {
-        return AotHelper.IsAot
-            ? AotSerializeToUtf8Bytes()
-            : JitSerializeToUtf8Bytes();
-
+#if NET8_0_OR_GREATER
```

Comment:
@vaind I think this implicitly means we need to change when we support AOT compilation:
https://github.com/getsentry/sentry-dotnet/blob/ebaed3e131d7346ce15ad28a11d37bcd30a99b0a/src/Directory.Build.props#L38-L41

If the serialization only works on net8.0 or greater then the framework only supports AOT on net8.0 or greater as well right (due to trimming)?

Reply:
Ah that's right AOT is around since .NET 7, .NET 8 added macOS support

Reply:
This condition must be for net8 because of a new API it uses.

Reply:
Good to know we have net8-only code already to validate the need to bump it.
That alone is a reason to add net8 to main packages that can be added alone (and hence bringing the net8 version of Sentry)